### PR TITLE
Demonstrate how to get good ghosted solutions from a ExternalProblem

### LINKED ISSUE #17534: AuxKernel required for proper Postprocessor evaluations with ExternalProblem
## Bug Description
When using the `ExternalProblem` interface, postprocessors of variables added via `addExternalVariables` and written in `syncSolutions` are lagged by one time step if the input file lacks _any_ AuxKernels. For our OpenMC wrapping, our input files are quite slim, and don't have any AuxVariables or AuxKernels defined in the XML files. For the first time step, the lack of an AuxKernel is causing our `ElementIntegralVariablePostprocessor` of an external variable that we write to within `syncSolutions` to evaluate to 0.0, even though the variable is displayed correctly in the Exodus output and is nonzero.

## Steps to Reproduce
I have attached a minimum working example with a very slimmed down `BasicExternalProblem` that creates a CONSTANT MONOMIAL variable `heat_source` and writes a value of `12345.0` to it within `syncSolutions`. If the input file looks like this:

```
[Mesh]
  type = GeneratedMesh
  dim = 2
[]

[Problem]
  type = BasicExternalProblem
[]

[AuxVariables]
  [dummy]
  []
[]

[Executioner]
  type = Transient
  num_steps = 1
[]

[Postprocessors]
  [heat_source]
    type = ElementIntegralVariablePostprocessor
    variable = heat_source
  []
[]

[Outputs]
  exodus = true
[]
```

then even if we are writing correctly to `heat_source` (and the Exodus file shows a heat source of 12345.0), the postprocessor evaluates to zero:

```
Time Step 1, time = 1, dt = 1
Transfer FROM app, setting value of 12345.0
 Solve Converged!

Postprocessor Values:
+----------------+----------------+
| time           | heat_source    |
+----------------+----------------+
|   0.000000e+00 |   0.000000e+00 |
|   1.000000e+00 |   0.000000e+00 |
+----------------+----------------+
```

However, if you just add a dummy AuxKernel that has nothing to do with the physics, i.e. adding this to the above file:

```
# Need to add this to see the evaluation of the heat source postprocessor
[AuxKernels]
  [dummy]
    type = ConstantAux
    variable = dummy
    value = 0.0
  []
[]
```

then the postprocessor evaluates correctly:

```
Time Step 1, time = 1, dt = 1
Transfer FROM app, setting value of 12345.0
 Solve Converged!

Postprocessor Values:
+----------------+----------------+
| time           | heat_source    |
+----------------+----------------+
|   0.000000e+00 |   0.000000e+00 |
|   1.000000e+00 |   1.234500e+04 |
+----------------+----------------+
```

If you increase `num_steps`, you'll see that the values are _lagged_ by one time step for some reason. 

### What I think is the problem
So it seems that something needs to be fixed along the lines of:

The absence of auxkernels should not be what triggers when `_u` or `coupledValue` are evaluated (i.e. whatever Postprocessors grab) when using `ExternalProblem`.

## Impact
Annoyance - we can just add dummy AuxVariables and AuxKernels to all our inputs to get around this, but this is prone to errors for anyone who doesn't understand that dummy AuxKernels are needed to get correct problem output.

### PR DESCRIPTION
**Pull request by lindsayad**
We cannot simply update the parallel solution. We have to do a second step of localizing the parallel solution to the ghosted current local solution

Closes #17534

### CODE REVIEW

**Comment by lindsayad** [comment]
@aprilnovak

File: test/include/problems/TransientExternalProblem.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (GiudGiud) [inline test/include/problems/TransientExternalProblem.h:1]: missing headers in these files

File: test/src/problems/TransientExternalProblem.C [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#include "TransientExternalProblem.h"
+#include "AuxiliarySystem.h"
+
+registerMooseObject("MooseTestApp", TransientExternalProblem);
```
Comment (GiudGiud) [inline test/src/problems/TransientExternalProblem.C:4]: I would change the name. (SyncTestExternalProblem?)
The name would be great for pointing to specificities of a transient external problem

but there's no specificity here, you re just using time to get a good test output

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Review by GiudGiud — DISMISSED** [review]

File: test/src/problems/SyncTestExternalProblem.C [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      break;
+    case ExternalProblem::Direction::FROM_EXTERNAL_APP:
+    {
+      auto & solution = _aux->solution();
+      const auto sys_number = _aux->number();
+      const auto & mesh = _mesh.getMesh();
+
+      for (unsigned int e = 0; e < mesh.n_elem(); ++e)
+      {
+        auto elem_ptr = mesh.query_elem_ptr(e);
```
Comment (loganharbour) [inline test/src/problems/SyncTestExternalProblem.C:57]: Element range loop?

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by loganharbour — APPROVED** [review]
No more issues from me - simple and to the point.

**Comment by lindsayad** [comment]
failure unrelated

**Pull request merged**